lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+    'EMEA-FR-PAR1': {'name': 'Paris 1', 'country': 'FR', "api_endpoint": 'https://cp-par1.scaleway.com'},
+    'ams1': {'name': 'Amsterdam 1', 'country': 'NL', "api_endpoint": 'https://cp-ams1.scaleway.com'},
+    'EMEA-NL-EVS': {'name': 'Amsterdam 1', 'country': 'NL', "api_endpoint": 'https://cp-ams1.scaleway.com'},
+}
+
+SCALEWAY_SERVER_STATES = (
+    'stopped',
+    'stopping',
+    'starting',
+    'running',
+    'locked'
+)
+
+SCALEWAY_TRANSITIONS_STATES = (
+    "stopping",
+    "starting",
+    "pending"
+)
+
+
+def print_warning(compute_api, msg):
+    compute_api.module.warn("%s %s" % (datetime.datetime.utcnow(), msg))
+
+
+def fetch_state(compute_api, server):
+    print_warning(compute_api, "fetch_state of server: %s" % server["id"])
+    response = compute_api.get(path="servers/%s" % server["id"])
+
+    if response.status_code == 404:
+        return "absent"
+
+    if not response.ok:
+        msg = 'Error during state fetching: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    try:
+        print_warning(compute_api=compute_api, msg="Server %s in state: %s" % (server["id"], response.json["server"]["state"]))
+        return response.json["server"]["state"]
+    except KeyError:
+        compute_api.module.fail_json(msg="Could not fetch state in %s" % response.json)
+
+
+def wait_to_complete_state_transition(compute_api, server):
+    wait_timeout = compute_api.module.params["wait_timeout"]
+    wait_sleep_time = compute_api.module.params["wait_sleep_time"]
+
+    start = datetime.datetime.utcnow()
+    end = start + datetime.timedelta(seconds=wait_timeout)
+    while datetime.datetime.utcnow() < end:
+        print_warning(compute_api, "We are going to wait for the server to finish its transition")
+        if fetch_state(compute_api, server) not in SCALEWAY_TRANSITIONS_STATES:
+            print_warning(compute_api, "It seems that the server is not in transition anymore.")
+            print_warning(compute_api, "Server in state: %s" % fetch_state(compute_api, server))
+            break
+        time.sleep(wait_sleep_time)
+    else:
+        compute_api.module.fail_json(msg="Server takes too long to finish its transition")
+
+
+class Response(object):
+
+    def __init__(self, resp, info):
+        self.body = None
+        if resp:
+            self.body = resp.read()
+        self.info = info
+
+    @property
+    def json(self):
+        if not self.body:
+            if "body" in self.info:
+                return json.loads(self.info["body"])
+            return None
+        try:
+            return json.loads(self.body)
+        except ValueError:
+            return None
+
+    @property
+    def status_code(self):
+        return self.info["status"]
+
+    @property
+    def ok(self):
+        return self.status_code in (200, 201, 202, 204)
+
+
+class ScalewayComputeAPI(object):
```

Comment:
This class should be moved to `module_utils/scaleway.py` and reused by `scaleway_sshkey` module.